### PR TITLE
Drag and drop

### PR DESCRIPTION
**Pull request by marco9446**
Added the possibility to drag and drop an element from the right nav bar into the slide canvas.
All the imported element are lazy loaded and injected in the slide area.
The drop target is now hard coded to the rood of slide area (paper-element) but it already prepared to work with a customisable target

### CODE REVIEW

**Comment by triglian** [comment]
### Tests

When testing an element (unit-test) you should minimally test
1. Public methods
2. setting and getting public properties
3. content nodes (check if content is distributed like you want it to)
4. render (if sub elements are where they need to be, if dom-if templates work as they should etc.)

When testing a feature normally there's more than a few elements involved. There you need to programmatically perform a user interaction (e.g drag and drop) and check if everything is at it's place. Bear in mind that how an element renders is also important

Check the designer for tests

**Comment by marco9446** [comment]
I changed the name of components in order to follow convention and avoid name collision.
I did't find a linter for javascript, compatible with polymer and gulp, so i just implemented polylint.
The droppable are is now the whole stage and i made some changes in the style to remain as neutral as possible and you have also the ability to create group in the draggable list.
I'll make some test after having converted the project to ES6

**Comment by marco9446** [comment]
I have decided to abandon the idea of using a custom element to wrap selected item since i haven't found any good reason to proceed in that direction.
Right now I just wrap the dynamically created elements in a DIV

File: app/elements/qea-property-analyzer/qea-property-analyzer.html [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                reject(new Error('impossible to analyze element '+ this.src))
+              }
+              resolve(this._analyzer)
+            })
+          }
+        )
+      }
+
+      // return the list of behaviors of the element analyzed
+      getBehaviors(analyzer){
```
Comment (triglian) [inline app/elements/qea-property-analyzer/qea-property-analyzer.html]: You have a getter for behaviours and then you have the `getBehaviors` function. What's their difference? As an interface, isn't this confusing?

File: app/elements/qea-property-analyzer/qea-property-analyzer.html [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+         transitive=[[recursively]]
+         loading={{loading}}
+         clean>
+    </hydrolysis-analyzer>
+
+  </template>
+  <script>
+  (function () {
+    'use strict'
+
```
Comment (triglian) [inline app/elements/qea-property-analyzer/qea-property-analyzer.html:26]: I assume all the properties and methods that don't start with an `_` are supposed to be accessible publicly right?

Make sure this is the case for all files

File: app/elements/qea-question-type/qea-question-type.html [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        }
+      }
+
+      _startDragging(){
+        this.isDragging = 'isDragging';
+        this.fire('dragStart')
+      }
+
+      _currentlyDragging(dragDetail){
+         // console.log(dragDetail);
```
Comment (triglian) [inline app/elements/qea-question-type/qea-question-type.html]: remove the loggint statement.

File: app/test/qea-property-analyzer-test.html [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          myEl = fixture('qea-property-analyzer-fixture');
+          stub('qea-property-analyzer', {
+            resolvePath: function(tagName) {
+              return '../../.tmp/bower_components/'+tagName+'/'+tagName+'.html'
+            }
+          });
+        });
+
+
+        test('test scanElemen method', function(done) {
```
Comment (triglian) [inline app/test/qea-property-analyzer-test.html]: Your don't need "check" or  "test" on your test messages, it is implied.  You should try to thing of the message like :
(test that the) "scanElement method succeeds for asq elements'".  Check here for some examples: https://github.com/Polymer/designer/blob/master/test/path/path_test.html. Please change all affected messages across all test files.

File: electron.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+'use strict';
```
Comment (triglian) [inline electron.js:1]: If you borrowed this from somewhere, it is good if you mention it on the top of the file.

File: app/elements/qea-property-analyzer/qea-property-analyzer.html (hunk 1 above)
Comment (marco9446) [inline app/elements/qea-property-analyzer/qea-property-analyzer.html]: The getter for behaviours is a polymer notation to inject behaviours inside that component instead the getBehaviors function return the behaviours use by the element analysed by hydrolysis .

File: electron.js (hunk 5 above)
Comment (marco9446) [inline electron.js:1]: I took it from the [Quick Start guide](http://electron.atom.io/docs/latest/tutorial/quick-start/) of electron so i don't think it's necessary to mention

File: app/elements/qea-property-analyzer/qea-property-analyzer.html (hunk 2 above)
Comment (marco9446) [inline app/elements/qea-property-analyzer/qea-property-analyzer.html:26]: fixed

File: app/elements/qea-property-analyzer/qea-property-analyzer.html (hunk 1 above)
Comment (triglian) [inline app/elements/qea-property-analyzer/qea-property-analyzer.html]: then maybe `getBehaviors` -> `getElementBehaviors`?

File: bower.json [hunk 6]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "editor-project",
```
Comment (triglian) [inline bower.json]: the name is `question-editor-asq`

File: app/test/qea-property-analyzer-test.html [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+          myEl = fixture('qea-property-analyzer-fixture');
+          stub('qea-property-analyzer', {
+            resolvePath: function(tagName) {
+              return '../../.tmp/bower_components/'+tagName+'/'+tagName+'.html'
+            }
+          });
+        });
+
+
+        test('retrun an analyzer object from scanElement method', function(done) {
```
Comment (triglian) [inline app/test/qea-property-analyzer-test.html]: `retrun`

**Comment by triglian** [comment]
There should be a command that runs the test when you do `npm test`

**Pull request merged**